File: ms_cinema/domain/model/src/main/java/co/com/luisgomez29/model/common/exception/TechnicalException.java
package co.com.luisgomez29.model.common.exception;


import co.com.luisgomez29.model.common.enums.TechnicalExceptionMessage;
import lombok.Getter;

@Getter
public class TechnicalException extends RuntimeException {

    private final TechnicalExceptionMessage technicalExceptionMessage;

    public TechnicalException(TechnicalExceptionMessage technicalExceptionEnum) {
        super(technicalExceptionEnum.getMessage());
        this.technicalExceptionMessage = technicalExceptionEnum;
    }

    public TechnicalException(String message, TechnicalExceptionMessage technicalException) {
        super(message);
        this.technicalExceptionMessage = technicalException;
    }

    public TechnicalException(Throwable cause, TechnicalExceptionMessage message) {
        super(cause);
        this.technicalExceptionMessage = message;
    }
}

File: ms_cinema_kt/domain/model/src/main/kotlin/co/com/luisgomez29/model/exception/TechnicalException.kt
package co.com.luisgomez29.model.exception

import co.com.luisgomez29.model.enums.TechnicalExceptionMessage


class TechnicalException : RuntimeException {
    private val technicalExceptionMessage: TechnicalExceptionMessage

    constructor(technicalExceptionEnum: TechnicalExceptionMessage) : super(technicalExceptionEnum.name) {
        this.technicalExceptionMessage = technicalExceptionEnum
    }

    constructor(message: String?, technicalExceptionMessage: TechnicalExceptionMessage) : super(message) {
        this.technicalExceptionMessage = technicalExceptionMessage
    }

    constructor(cause: Throwable?, message: TechnicalExceptionMessage) : super(cause) {
        this.technicalExceptionMessage = message
    }
}